idoc/bilqr.py

```diff
@@ -52,23 +52,26 @@ def make_lqr_approx(cs: Problem, params: Params, local=True) -> Callable:
     """Create LQR approximation function"""
     T = cs.horizon
     x0, theta = params.x0, params.theta
+    vm = lambda f: vmap(f, in_axes=(None, 0, None, None))
 
-    @jax.vmap
+    @vmap
     def approx_timestep(t, x, u):
-        #here : add the vmaps : x is of size BxN
-        M = jax.vmap(jax.jacfwd(jax.grad(cs.cost, argnums=2), argnums=1),in_axes=(None,0,None,None))(t, x, u, theta)
-        R = jnp.sum(jax.vmap(jax.jacfwd(jax.grad(cs.cost, argnums=2), argnums=2), in_axes=(None,0,None,None))(t, x, u, theta),axis=0)
-        q = jax.vmap(jax.grad(cs.cost, argnums=1), in_axes = (None,0,None,None))(t, x, u, theta)
-        Q = jax.vmap(jax.jacfwd(jax.grad(cs.cost, argnums=1), argnums=1),in_axes=(None,0,None,None))(t, x, u, theta)
-        r = jax.vmap(jax.grad(cs.cost, argnums=2), in_axes = (None,0,None,None))(t, x, u, theta)
-        A = jax.vmap(jax.jacobian(cs.dynamics, argnums=(1)),in_axes = (None,0,None,None))(t, x, u, theta)
-        B = jax.vmap(jax.jacobian(cs.dynamics, argnums=2),in_axes = (None,0,None,None))(t, x, u, theta)
+        # here : add the vmaps : x is of size BxN
+        M = vm(jacfwd(grad(cs.cost, argnums=1), argnums=2))(t, x, u, theta)
+        R = vm(jacfwd(grad(cs.cost, argnums=2), argnums=2))(t, x, u, theta).sum(0)
+        q = vm(grad(cs.cost, argnums=1))(t, x, u, theta)
+        Q = vm(jacfwd(grad(cs.cost, argnums=1), argnums=1))(t, x, u, theta)
+        r = vm(grad(cs.cost, argnums=2))(t, x, u, theta).sum(0)
```

Comment:
summing should indeed be fine because we only ever need the sum over examples (R is identical for all examples but r is data-dependent)